### ISSUE TITLE
Clash between Final and consider-using-assignment-expr

### ISSUE THREAD

**Issue by nth10sd** [pylint-dev/pylint#9079]
### Bug description

Testcase:

```python
import shutil
from typing import Final

AA: Final = shutil.which("aa")
if not AA:
    raise FileNotFoundError
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
R6103: Use 'if not (AA := shutil.which('aa')):' instead (consider-using-assignment-expr)
```

### Expected behavior

There is no way to declare a variable as `Final` in an assignment expression, so if I were to follow pylint's recommendation, I cannot set the variable as `Final`. There should be no error.

### Pylint version

```shell
pylint 2.17.6
astroid 2.15.7
Python 3.10.13
```

### OS / Environment

Ubuntu 22.04

**Comment by nth10sd** [comment]
Please let me know if this is working as intended and that I should use an ignore line. For reference, other tools such as `ruff` do not complain about this.

**Comment by Pierre-Sassoulas** [comment]
You can always disable the message locally using https://pylint.readthedocs.io/en/stable/user_guide/messages/message_control.html#block-disables. 

PEP0572 suggest this:
```python
# Inline type annotations are not supported:

# Closest equivalent is "p: Optional[int]" as a separate declaration
p: Optional[int] = None
```

https://peps.python.org/pep-0572/#differences-between-assignment-expressions-and-assignment-statements

I classified that as false positive originally but waiting for more opinions.

**Comment by jacobtylerwalls** [comment]
Agree that you can do this, so I don't think there's an inconsistency:
```py
AA: Final
if not (AA := shutil.which('aa')):
...
```
> For reference, other tools such as ruff do not complain about this.

Right, this is an "R" level message for "refactoring", so it's not that the code is wrong by any means, they're just a matter of taste.
